subsys/net/l2/ethernet/ethernet.c

```diff
@@ -181,7 +181,8 @@ enum net_verdict ethernet_check_ipv4_bcast_addr(struct net_pkt *pkt,
 	if (net_eth_is_addr_broadcast(&hdr->dst) &&
 	    !(net_ipv4_is_addr_mcast((struct in_addr *)NET_IPV4_HDR(pkt)->dst) ||
 	      net_ipv4_is_addr_bcast(net_pkt_iface(pkt),
-				     (struct in_addr *)NET_IPV4_HDR(pkt)->dst))) {
+				     (struct in_addr *)NET_IPV4_HDR(pkt)->dst)) &&
+	    !IS_ENABLED(CONFIG_NET_ACCEPT_MISMATCH_L3_L2_ADDR)) {
```

Comment:
No strong push, but I'd just return early, this complex if condition combo is a bit hard to read:
```C
if (IS_ENABLED(CONFIG_NET_ACCEPT_MISMATCH_L3_L2_ADDR)) {
	return NET_OK;
}
```